### PR TITLE
Feature/more special chars attempt#2

### PR DESCRIPTION
**Pull request by Klako**
Mirrored the code in [Google-Scoutnet-sync](https://github.com/scouternasetjanster/Google-Scoutnet-synk) to generate email-friendly usernames to be used in firstlast attribute.

#worksforme
pls try it.

### CODE REVIEW

File: lib/Auth/Source/scoutnetauth.php [hunk 1]
```diff
@@ -109,4 +108,126 @@ protected function login($username, $password)
...
+    /**
+     * Formats a name to be used in an email address.
+     * @param string $name
+     * @return string
+     */
+    private function formatNameForEmail($name)
+    {
+        $name = trim(strtolower($name)); // Convert to lower case and trim spaces.
+        $name = preg_replace('/\s+/', '.', $name); // Replace empty space with one dot each.
+        $name = preg_replace('/[.][\-]/', '-', $name); // Replace .- with -
```
Comment (puggan) [inline lib/Auth/Source/scoutnetauth.php:121]: why use character-groups with just one char in it?
'/[.][\\-]/' = '/\\.-/'
And why use regexp when there are just static strings?
`strtr($name, ['.-' => '-', '-.' => '-']);`

File: lib/Auth/Source/scoutnetauth.php [hunk 2]
```diff
@@ -109,4 +108,126 @@ protected function login($username, $password)
...
+        $name = preg_replace('/[\-][.]/', '-', $name); // Replace -. with -
+        $name = $this->removeDiacritics($name); // Replace special characters with normal ones.
```
Comment (puggan) [inline lib/Auth/Source/scoutnetauth.php:123]: strtolower should be done after reomveDiacritics, or name like "Åsa" becomes "Asa" insted of "asa".
ref: https://3v4l.org/SLcV7

File: lib/Auth/Source/scoutnetauth.php (hunk 1 above)
Comment (Klako) [reply lib/Auth/Source/scoutnetauth.php:121]: That's what [Google-Scoutnet-sync](https://github.com/scouternasetjanster/Google-Scoutnet-synk/blob/68e9bd9669eb7abc3c526a761a003b0105a796b6/Anvandare.gs#L258) does.
I'll change it with the bug you mentioned.

File: lib/Auth/Source/scoutnetauth.php (hunk 2 above)
Comment (puggan) [reply lib/Auth/Source/scoutnetauth.php:123]: For reference:
The javascript-version can do .toLowerCase() first, as it converts "Å"->"å", phps strtolower() dosn't convert "Å".

**Pull request merged**